src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4868,12 +4768,12 @@ private static NamespaceOrTypeSymbol ToLanguageSpecific(INamespaceOrTypeSymbol c
                 return null;
             }
 
-            var result = container as NamespaceOrTypeSymbol;
-            if ((object)result == null)
+            if (!(container is NamespaceOrTypeSymbol namespaceOrTypeSymbol))
```

Comment:
Nit: Consider swapping this condition around (using positive test instead of negative).